### DISCUSSION TITLE
Using custom models within Sharepoint site

### DISCUSSION THREAD

**Discussion by thorsted** [Smithsonian/dpo-voyager#130] [Q&A]
Hello,

Sharepoint has a method for embedding iframes, so it is fairly easy to link a Smithsonian model to my sharepoint site, but I don't see a path to adding my own models as the options available require html code. Has anyone done this?

**Comment by gjcope** [comment]
Hi @thorsted I'm not a regular Sharepoint user, but if iframes are an option it is still doable. Are you able to upload a page to a webserver anywhere? If so, just copy and paste the code under "Option 2" at: https://smithsonian.github.io/dpo-voyager/explorer/usage/ and modify it to point at your scene. Then use this url for the Sharepoint iframe.

**Comment by thorsted** [comment]
Was hoping to upload all content to a the sharepoint server. Sharepoint uses aspx and doesn't play well with regular HTML code and javascript, hence my problem.

**Comment by gjcope** [comment]
Since Voyager is a web component, unfortunately the html tag has to exist somewhere. Hopefully someone with SP experience can weigh in, but this: https://answers.microsoft.com/en-us/msoffice/forum/all/how-to-embed-html-code-into-sharepoint/9409b31f-00ce-4b55-abb1-03897403bb37 seems to imply you can just rename a page to aspx and upload? The page would be quite simple so it might be worth a try.

**Reply by thorsted** [reply]
I have tried many methods of doing this, but I think the issue is the required javascript, sharepoint doesn't seem to allow it.

**Comment by gjcope** [comment]
Could you share the errors/issues you are seeing with Sharepoint loading the page?

**Reply by thorsted** [reply]
The iframe will just show a black box, nothing loads. Or the aspx or html file goes directly to download. Hard to really say what is happening. Might be the javascript, as sharepoint doesn't allow injecting of javascript or the direct links to the content folder and json file get missed because of the sharepoint file structure.

**Comment by gjcope** [comment]
Can you share the output of your browser's developer console?
